### PR TITLE
[QF-3991] Redesign Auth UI

### PR DESCRIPTION
**Pull request by al-imam**
## Summary

This PR implements redesign of the authentication flow, simplifying the user experience by removing the multi-step social/email selection process and defaulting to email login.

Closes: [QF-3991](https://quranfoundation.atlassian.net/browse/QF-3991)

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] ♻️ Refactoring (no functional changes)

## Environment Variables
No new environment variables introduced.

## Test Plan

- [ ] Unit tests added/updated
- [x] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Navigate to /login page
2. Verify email form is displayed by default (no social login step)
3. Test sign in functionality with valid credentials
4. Test sign up functionality with new user
5. Test forgot password flow
6. Test social login buttons still work
7. Verify responsive design and RTL layout

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [ ] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [x] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [x] Inline comments added for complex logic

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [x] RTL layout verified

### Accessibility (if UI changes)

- [x] Semantic HTML elements used
- [x] ARIA attributes added where needed
- [x] Keyboard navigation works

## Screenshots/Videos

| Before | After |
| ------ | ----- |
| image | image |

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code (Mostly used for review)

[QF-3991]: https://quranfoundation.atlassian.net/browse/QF-3991?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/styles/theme.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
-@use "./themes/light";
-@use "./themes/dark";
-@use "./themes/sepia";
-@use "./shades";
+@use './themes/light';
+@use './themes/dark';
+@use './themes/sepia';
+@use './shades';
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/styles/theme.scss:4]:
```JSON
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "bracketSameLine": false,
  "printWidth": 100,
  "proseWrap": "always",
  "requirePragma": false,
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all",
  "useTabs": false
}
```

Prettierrc config has `"singleQuote": true` so my formating is right if needed to change formating then we have to change `singleQuote` to `false`.

Marking as Resolved.

File: src/components/Login/SignUpForm/PasswordInput.module.scss [hunk 2]
```diff
@@ -1,60 +1,72 @@
...
-        color: var(--color-text-default);
-    }
+  position: relative;
+  background: none;
+  border: none;
+  padding: 4px;
+  margin-top: var(--spacing-xxsmall-px /* 5px */);
+  cursor: pointer;
+  color: var(--color-text-faded);
+  padding-inline: var(--spacing-xxsmall /* 6px */);
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Login/SignUpForm/PasswordInput.module.scss]: I added these values because I reviewed the PR with multiple AIs, all of which said to use '--spacing-xxsmall' and remove '--spacing-xxsmall-px'. This is to make it clear that these CSS variables do not have the same value.

Removing it ✅

File: src/components/Login/login.module.scss [hunk 3]
```diff
@@ -365,90 +286,72 @@ $container-width: 370px;
...
 }
 
-.servicesContainer {
+.subtitleAndLogoGroup {
   display: flex;
   flex-direction: column;
   align-items: center;
+  gap: var(--spacing-xsmall);
+  margin-top: var(--spacing-xxlarge-px);
+  margin-bottom: var(--spacing-large-px);
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Login/login.module.scss]: Using logical properties

File: src/components/Login/login.module.scss [hunk 4]
```diff
@@ -289,6 +223,10 @@ $container-width: 370px;
   }
 }
 
+.smallMarginTop {
+  margin-top: var(--spacing-small-px);
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Login/login.module.scss]: Using logical properties

File: src/components/Login/login.module.scss [hunk 5]
```diff
@@ -41,76 +31,41 @@ $container-width: 370px;
...
 .authButtons {
   display: flex;
-  flex-direction: column;
+  flex-direction: row;
   gap: var(--spacing-medium);
   width: 100%;
+
+  & > * {
+    flex: 1;
+  }
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Login/login.module.scss:46]: Align with the Figma design and has enough space to click

File: locales/en/login.json [hunk 6]
```diff
@@ -81,30 +75,21 @@
...
-  "reflect-feature-2": "Join Groups",
-  "reflect-feature-3": "Interact with others",
-  "reflect-feature-4": "And more!",
+
   "reset-password": "Reset Password",
   "reset-password-success": "Password reset successfully!",
   "set-new-password": "Set a new password",
   "sign-in": "Sign in",
-  "sign-in-or-sign-up": "Sign in or Sign up",
+  "sign-in-or-sign-up": "Sign In or Sign Up below to continue to",
```
[the remark this answers is not in the document]
Comment (al-imam) [reply locales/en/login.json:83]: The text is directly from Figma. It has also been verified in the Jira ticket and is correct.
